doc/source/whatsnew/v1.0.0.rst

```diff
@@ -656,6 +656,7 @@ Other
 - :meth:`Series.append` will no longer raise a ``TypeError`` when passed a tuple of ``Series`` (:issue:`28410`)
 - :meth:`SeriesGroupBy.value_counts` will be able to handle the case even when the :class:`Grouper` makes empty groups (:issue: 28479)
 - Fix corrupted error message when calling ``pandas.libs._json.encode()`` on a 0d array (:issue:`18878`)
+- Backtick quoting in :meth:`DataFrame.query` and :meth:`DataFrame.eval` can now also be used to use invalid identifiers like names that start with a digit, are python keywords, or are using single character operators. (:issue:`27017`)
```

Comment:
is pd.eval affected?

Reply:
No, DataFrame.eval makes use of pd.eval, but that is downstream in de code of the changes made in this PR. pd.eval expects a query and some name resolvers. This PR interprets the backtick quotes and replaces them with Python valid identifiers in addition to updating the name resolvers, before sending them downstream and then eventually to pd.eval. But pd.eval is not affected by this.